blocks/api/factory.js

```diff
@@ -17,10 +17,20 @@ import { getBlockSettings } from './registration';
  * @return {Object}             Block object
  */
 export function createBlock( blockType, attributes = {} ) {
+	const blockSettings = getBlockSettings( blockType );
+
+	let defaultAttributes;
+	if ( blockSettings ) {
+		defaultAttributes = blockSettings.defaultAttributes;
+	}
+
 	return {
 		uid: uuid(),
 		blockType,
-		attributes
+		attributes: {
+			...defaultAttributes,
```

Comment:
>Could we use `blockSettings.defaultAttributes` directly here. You showed me that `Object.assign` handles the `undefined` value properly :)

It'll handle an undefined `defaultAttributes` here, but if `blockSettings` doesn't exist and we try to access the `defaultAttributes` property on it, an error would be thrown.